hbase-server/src/main/java/org/apache/hadoop/hbase/replication/ChainWALEntryFilter.java

```diff
@@ -68,13 +82,17 @@ public void initCellFilters() {
 
   @Override
   public Entry filter(Entry entry) {
+
     for (WALEntryFilter filter : filters) {
       if (entry == null) {
         return null;
       }
       entry = filter.filter(entry);
     }
     filterCells(entry);
+    if (shouldFilterEmptyEntry() && entry != null && entry.getEdit().isEmpty()) {
```

Comment:
Put this into your own WALEntryFilter. Or, fine if you want to add a WALEntryFilter impl to HBase code that does this. But don't do this here

Reply:
@apurtell This is the flexility we want to provide to customer replication endpoint so that they every custom replication endpoint doesn't need to re-implement everything what `ChainWALEntryFilter` already does. 

Here is the scenario we want to cover, let's take an example:
```java
CustomWALFilter implements WALEntryFilter, WALCellFilter {
    @override 
     public void filter(Entry) {
    }
    @override 
    public void filterCell(Entry, Cell){
    }
}
```
Custom Replication endpoint set the filters by:
```java
ChainWALEntryFilter(filters); \\ new CustomWALFilter() is part of filters
```
if `filter` in the above `CustomWALFilter` returns Entry but filterCell filters all the cells, the `ChainWALEntryFilter` will not return null in current implementation. Since most of the `CustomWALFilter` uses `ChainWALEntryFilter`, isn't it better to provide this flexibility in ChainWALFilter itself? Sure, it's possible to all this logic in CustomWALFilter, but shouldn't we provide re-usability through ChainWALFilter and provides a flexibility through a config? Let me know your thoughts.

Reply:
I think this additional logic would fit better in a built-in filter (one of @apurtell suggestions above). That would not need the extra config property nor changes in _BaseRepplicationEndpoint_